src/model/engines/engine_base.nit

```diff
@@ -53,5 +76,79 @@ class Engine
 	fun compile(submission: Submission): Bool is abstract
 
 	# Run `test` for `submission`
-	fun run_test(submission: Submission, test: TestCase): TestResult is abstract
+	fun run_test(submission: Submission, test: TestCase): TestResult do
+		var res = new TestResult(test, submission)
+
+		var tdir = "test{submission.results.length + 1}"
```

Comment:
They are just directory names the fronted (nor the model) does not even show them.
However, we might need to title (number? name?) the tests if we want to present them to the player.